pkg/storegateway/indexheader/fileutil/mmap_windows.go

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/tsdb/fileutil/mmap_windows.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors.
+
```

Comment:
It apparently compiles, but shouldn't this file build on Windows only?

```suggestion

//go:build windows
// +build windows

```

Reply:
@pstibrany ah, that must be a rule I've forgot exists.

Reply:
Your question reminded me about it :) I checked original code in Prometheus, and it also doesn't have build instructions.